File: Unit-3/week2/Monday/DynamicFragments/app/src/main/java/com/example/dynamicfragments/Network.java
package com.example.dynamicfragments;


import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Network {

    private static HttpLoggingInterceptor httpLoggingInterceptor = new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY);

    public static retrofit2.Retrofit getInstance(Class<ApiService> apiServiceClass) {
        Retrofit.Builder builder = new Retrofit.Builder().baseUrl("")
                .addConverterFactory(GsonConverterFactory.create())
                .client(new OkHttpClient.Builder()
                        .addInterceptor(httpLoggingInterceptor)
                        .build());
        return builder.build();
    }

}

File: Unit-3/week4/Day3/ScopeFunctions/app/src/main/java/com/example/scopefunctions/Network.kt
package com.example.scopefunctions

import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit

class Network {

    companion object{

        private val httpLoggingInterceptor = HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY);


        fun getInstance() : Retrofit{

            return Retrofit.Builder()
                .baseUrl("https:/reqres.in/")
                .addConverterFactory()
        }

    }
}